htlcswitch/interfaces.go

```diff
@@ -110,6 +110,10 @@ type ChannelLink interface {
 	// will use this function in forwarding decisions accordingly.
 	EligibleToForward() bool
 
+	// ChannelState returns a pointer to a channeldb.OpenChannel representing
+	// the state of the channel.
+	ChannelState() *channeldb.OpenChannel
```

Comment:
I don't think we need to expose underlying details like this, it effectively weakens the abstraction. Instead, the switch only needs to tell the channel notifier of a channel point, and the notifier can gain a new specific interface which allows it to fetch a channel by its outpoint.

Reply:
fixed -- now we expose `ChannelPoint()` however